swift/example_code/sqs/scenario/Sources/entry.swift

```diff
@@ -0,0 +1,716 @@
+        let numOptions = options.count
+
+        if numOptions == 0 {
+            return 0
+        }
+
+        print(prompt)
+
+        var index = 0
+        for option in options {
+            print("(\(index)) \(option)")
+            index += 1
+        }
+        print("")
+
+        repeat {
+            print("Enter your selection (0 - \(numOptions-1)): ", terminator: "")
+            if let answer = readLine() {
+                guard let answer = Int(answer) else {
+                    print("Please enter the number matching your selection.")
+                    continue
+                }
+
+                if answer >= 0 && answer < numOptions {
+                    return answer
+                } else {
+                    print("Please enter the number matching your selection.")
+                }
+            }
+        } while true
+    }
+    
+    /// Ask the user too press RETURN. Accepts any input but ignores it.
+    /// 
+    /// - Parameter prompt: The text prompt to display.
+    func returnRequest(prompt: String) {
+        print(prompt, terminator: "")
+        _ = readLine()
+    }
+
+    var attrValues = [
+        "<none>",
+        "cheerful",
+        "funny",
+        "serious",
+        "sincere"
+    ]
+
+    /// Ask the user to choose one of the attribute values to use as a filter.
+    /// 
+    /// - Parameters:
+    ///   - message: A message to display before the menu of values.
+    ///   - attrValues: An array of strings giving the values to choose from.
+    /// 
+    /// - Returns: The string corresponding to the selected option.
+    func askForFilter(message: String, attrValues: [String]) -> String? {
+        print(message)
+        for (index, value) in attrValues.enumerated() {
+            print("  [\(index)] \(value)")
+        }
+
+        var answer: Int?
+        repeat {
+            answer = Int(stringRequest(prompt: "Select an value for the 'tone' attribute or 0 to end: "))
+        } while answer == nil || answer! < 0 || answer! > attrValues.count + 1
+
+        if answer == 0 {
+            return nil
+        }
+        return attrValues[answer!]
+    }
+
+    /// Prompts the user for filter terms and constructs the attribute
+    /// record that specifies them.
+    /// 
+    /// - Returns: A mapping of "FilterPolicy" to a JSON string representing
+    ///   the user-defined filter.
+    func buildFilterAttributes() -> [String:String] {
+        var attr: [String:String] = [:]
+        var filterString = ""
+
+        var first = true
+
+        while let ans = askForFilter(message: "Choose a value to apply to the 'tone' attribute.",
+                                    attrValues: attrValues) {
+            if !first {
+                filterString += ","
+            }
+            first = false
+
+            filterString += "\"\(ans)\""
+        }
+
+        let filterJSON = """
+                        { "tone": [\(filterString)]}
+                        """
```

Comment:
Considering how short this JSON fragment is, maybe reduce it to a 1-line string?

Reply:
Done.